### PR TITLE
TIME_CLOCKSOURCE now accepts both types - hyperv_clocksource and hv_clocksource

### PR DESCRIPTION
**Pull request by juhlee-microsoft**
Changed the condition for broadening distros with its older versions.

Lazy to post 12 results here, but do the screenshot.
image

### CODE REVIEW

File: Testscripts/Linux/TIME-CLOCKSOURCE.sh [hunk 1]
```diff
@@ -27,16 +27,16 @@ CheckSource()
...
 	current_clocksource="/sys/devices/system/clocksource/clocksource0/current_clocksource"
 
-	# Microsoft LIS installed version has lis_hyperv_clocksource_tsc_page
-	# Without Microsoft LIS, hyperv_clocksource_tsc_page
-	# CentOS 6.8 or older versions, hyperv_clocksource
-	clocksource="hyperv_clocksource_tsc_page"
-	mj=$(echo "$DISTRO_VERSION" | cut -d '.' -f 1)
-	mn=$(echo "$DISTRO_VERSION" | cut -d '.' -f 2)
-	if [[ $DISTRO_NAME == "centos" || $DISTRO_NAME == "rhel" ]] && [[ $mj -lt 7 && $mn -lt 9 ]]; then
+	# Microsoft LIS installed version has lis_hyperv_clocksource_tsc_page,
```
Comment (santoshx) [inline Testscripts/Linux/TIME-CLOCKSOURCE.sh]: Please use 'or' instead of comma.

File: Testscripts/Linux/TIME-CLOCKSOURCE.sh [hunk 2]
```diff
@@ -117,25 +118,24 @@ function UnbindCurrentSource()
 #
 # MAIN SCRIPT
 #
-GetDistro
 case $DISTRO in
-	redhat_6 | centos_6 | redhat_7 | centos_7)
+	redhat_6 | centos_6 | redhat_7 | centos_7 | debian*)
```
Comment (santoshx) [inline Testscripts/Linux/TIME-CLOCKSOURCE.sh:124]: How come debian suddenly doesn't support unbind?

Comment (juhlee-microsoft) [reply Testscripts/Linux/TIME-CLOCKSOURCE.sh:124]: This is the same question I have. I looked into the change. The available_clocksource has 3 values "hyperv_clocksource_tsc_page hyperv_clocksource acpi_pm", but the test code looks for acpi_pm always but Debian presented hyperv_clocksource at the end. I don't think Debian doing the same pattern as other distros. The database shows we have the result with the customized kernel most of the time.  Even we ran it in Debian 9 latest image but used the custom kernel.  I propose here we set the test principle working in the latest Azure image, not the customized image or kernel. 
Does it sound good? @santoshx please let me know if you have any other questions. I asked your review as you did this part change last time.

File: Testscripts/Linux/TIME-CLOCKSOURCE.sh (hunk 1 above)
Comment (juhlee-microsoft) [reply Testscripts/Linux/TIME-CLOCKSOURCE.sh]: Fixed in 6e85d8bb0333e43cf3ad087c88b000fcfbbe74c5

**Review by santoshx — DISMISSED** [review]

File: Testscripts/Linux/TIME-CLOCKSOURCE.sh [hunk 3]
```diff
@@ -117,25 +118,24 @@ function UnbindCurrentSource()
...
 		LogMsg "WARNING: $DISTRO does not support unbind current clocksource, only check"
 		CheckSource
 		;;
 	redhat_8 |centos_8|fedora*|clear-linux-os)
 		CheckSource
 		UnbindCurrentSource
 		;;
-	ubuntu*|debian*)
+	ubuntu*)
```
Comment (abhimarathe) [inline Testscripts/Linux/TIME-CLOCKSOURCE.sh]: Can we combine rest of the cases with `| (or)` if we are doing same steps in each of them?

Comment (juhlee-microsoft) [reply Testscripts/Linux/TIME-CLOCKSOURCE.sh]: Fixed in 86558f1016853b7f4530044bc67f5cc65ee4adb2

**Review by abhimarathe — DISMISSED** [review]
Looks good to me. 
Please verify if tests pass with recent changes.

**Review by santoshx — DISMISSED** [review]

**Review by abhimarathe — APPROVED** [review]

**Review by johnsongeorge-w — APPROVED** [review]
The check for LIS will be simplified in other PR.

**Pull request merged**